src/controllers/controller.h

```diff
@@ -193,13 +196,33 @@ class ControllerJSProxy : public QObject {
             : m_pController(m_pController) {
     }
 
+    QHash<QString, QString> m_cache;
+
     // The length parameter is here for backwards compatibility for when scripts
     // were required to specify it.
     Q_INVOKABLE virtual void send(const QList<int>& data, unsigned int length = 0) {
         Q_UNUSED(length);
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QByteArray convertEncoding(const QString& targetCharset, const QString& value) {
```

Comment:
So I can expose enums to `QJSEngine` but C++ enum only accepts ints as value. Is there a way that I can use strings as value or modify the signature of this function to accepts either a QString or an enum?

Reply:
you can take a QJSValue (which acts as a QVariant) and then switch internally depending on the type of the contained value. I'm not sure if that complexity is worth it though. I would prefer to have an enum exposed. A string would make more sense if the set of encoding is open, but in this case its closed because we're limited by `QStringConverter::Encoding`.

Reply:
Oh no, since 6.4, the set is opened with `QStringConverter`. That's why my last commit uses `QTextCodec` prior to 6.4. Virtually any encoding supported by UCI should be available. See my last comment, I used Latin-9 which is not available in `QStringConverter::Encoding`.

Reply:
Actually, there is probably a little more we can do here. Apparently there are more optional codecs available using [`QStringConverter::availableCodecs()`](https://doc.qt.io/qt-6/qstringconverter.html#availableCodecs) which we might want to support. In this case we should probably mirror the `QStringEncoder` api a little and have a function with the enum and a function with the string. I wouldn't try to emulate that via the QJSValue overload stuff though because the manual dispatch is annoying. Instead I would create two different `Q_INVOKEABLE` methods that are implemented in terms of the correct `QTextEncoder` overload. I'd all add a another `Q_VERSION_CHECK`, so we can skip the UTF16 to UTF8 conversion for > Qt 6.8.

Reply:
> Oh no, since 6.4, the set is opened with `QStringConverter`. That's why my last commit uses `QTextCodec` prior to 6.4. Virtually any encoding supported by UCI should be available. See my last comment, I used Latin-9 which is not available in `QStringConverter::Encoding`.

Yeah, you're right. I didn't see that comment when I wrote the last one.